airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/connector_nightly_report.py

```diff
@@ -0,0 +1,118 @@
+from dagster import Output, asset, OpExecutionContext, MetadataValue
+import pandas as pd
+import json
+import os
+from orchestrator.ops.slack import send_slack_webhook
+from orchestrator.models.ci_report import ConnectorNightlyReport, ConnectorPipelineReport
+from orchestrator.config import (
+    NIGHTLY_COMPLETE_REPORT_FILE_NAME,
+)
+from orchestrator.templates.render import (
+    render_connector_nightly_report_md,
+)
+
+GROUP_NAME = "connector_nightly_report"
+
+# HELPERS
+
+
+def blob_to_model(blob, Model):
+    report = blob.download_as_string()
+    file_path = blob.name
+
+    # parse json
+    report_json = json.loads(report)
+
+    # parse into pydandic model
+    report_model = Model(file_path=file_path, **report_json)
+
+    return report_model
+
+
+def blobs_to_typed_df(blobs, Model):
+    # read each blob into a model
+    models = [blob_to_model(blob, Model) for blob in blobs]
+
+    # convert to dataframe
+    models_df = pd.DataFrame(models)
+
+    return models_df
+
+
+def get_latest_reports(blobs, number_to_get):
+    # We can sort by the name to get the latest 10 nightly runs
+    latest_nightly_complete_file_blobs = sorted(blobs, key=lambda blob: blob.name, reverse=True)[:number_to_get]
+    return latest_nightly_complete_file_blobs
+
+
+def get_relevant_test_outputs(latest_nightly_test_output_file_blobs, latest_nightly_complete_file_blobs):
```

Comment:
Is this the function retrieving connector specific test reports? 
I'd appreciate a docstring to explain the logic 😄